src/reader_async.rs

```diff
@@ -103,20 +107,30 @@ trait SharedIO<T: AsyncFileReader + Unpin + Clone + 'static> {
 
 #[wasm_bindgen]
 pub struct AsyncParquetFile {
-    reader: HTTPFileReader,
+    reader: ParquetObjectReader,
     meta: ArrowReaderMetadata,
     batch_size: usize,
     projection_mask: Option<ProjectionMask>,
 }
 
 impl SharedIO<HTTPFileReader> for AsyncParquetFile {}
+impl SharedIO<ParquetObjectReader> for AsyncParquetFile {}
 
 #[wasm_bindgen]
 impl AsyncParquetFile {
     #[wasm_bindgen(constructor)]
-    pub async fn new(url: String) -> WasmResult<AsyncParquetFile> {
-        let client = Client::new();
-        let mut reader = HTTPFileReader::new(url.clone(), client.clone(), 1024);
+    pub async fn new(url: String, options: Option<Object>) -> WasmResult<AsyncParquetFile> {
```

Comment:
Is using `Object` here better than using `JsValue`? Is that just for TS types?

Reply:
Interesting :thinking: apparently `OptionFromWasmAbi` is not defined for JsValue (but it is for Object) - I wondered why I did that :open_mouth: . Yeah, both necessary and *very* slightly more accurate (since JsValue includes things like numbers and so on).

Reply:
I see. Previously I took in `JsValue` but parsed it via `serde_wasm_bindgen` to `Option<OptionsStruct>`. https://github.com/geoarrow/geoarrow-rs/blob/c296ced466b34e5a9dd487058d34910f08738833/js/src/io/parquet/async.rs#L175-L176

I'm not sure if either way is better than the other. I think in that case where we have strict object types, we should be able to define the typescript code for the input though.